kani-driver/src/args.rs

```diff
@@ -377,29 +389,47 @@ impl KaniArgs {
         // TODO: these conflicting flags reflect what's necessary to pass current tests unmodified.
         // We should consider improving the error messages slightly in a later pull request.
         if natives_unwind && extra_unwind {
-            Err(Error::with_description(
+            return Err(Error::with_description(
                 "Conflicting flags: unwind flags provided to kani and in --cbmc-args.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else if self.cbmc_args.contains(&OsString::from("--function")) {
-            Err(Error::with_description(
+            ));
+        }
+        if self.cbmc_args.contains(&OsString::from("--function")) {
+            return Err(Error::with_description(
                 "Invalid flag: --function should be provided to Kani directly, not via --cbmc-args.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else if self.quiet && self.concrete_playback == Some(ConcretePlaybackMode::Print) {
-            Err(Error::with_description(
+            ));
+        }
+        if self.quiet && self.concrete_playback == Some(ConcretePlaybackMode::Print) {
+            return Err(Error::with_description(
                 "Conflicting options: --concrete-playback=print and --quiet.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else if self.concrete_playback.is_some() && self.output_format == OutputFormat::Old {
-            Err(Error::with_description(
+            ));
+        }
+        if self.concrete_playback.is_some() && self.output_format == OutputFormat::Old {
+            return Err(Error::with_description(
                 "Conflicting options: --concrete-playback isn't compatible with \
                 --output-format=old.",
                 ErrorKind::ArgumentConflict,
-            ))
-        } else {
-            Ok(())
+            ));
+        }
+        if self.concrete_playback.is_some() && self.jobs() != Some(1) {
+            // Concrete playback currently embeds a lot of assumptions about the order in which harnesses get called.
+            return Err(Error::with_description(
+                "Conflicting options: --concrete-playback isn't compatible with --jobs.",
+                ErrorKind::ArgumentConflict,
+            ));
         }
+        if self.jobs.is_some() && self.output_format != OutputFormat::Terse {
```

Comment:
It would be more convenient to automatically switch to terse mode (which we want to switch to anyway) when jobs is specified.

Reply:
I thought about it, but:

1. The real problem is there's nowhere to put that code. e.g. this function doesn't take `&mut` and shouldn't.
2. But I also felt like being annoying for the moment is fine until we stabilize, and I'm planning on fixing this issue soon as well.

Reply:
OK.